package_control/automatic_upgrader.py

```diff
@@ -237,15 +238,9 @@ def do_upgrades():
             # Wait so that the ignored packages can be "unloaded"
             time.sleep(0.7)
 
-            # We use a function to generate the on-complete lambda because if
-            # we don't, the lambda will bind to info at the current scope, and
-            # thus use the last value of info from the loop
-            def make_on_complete(name):
-                return lambda: self.installer.reenable_package(name)
-
             for info in package_list:
                 if info[0] in disabled_packages:
-                    on_complete = make_on_complete(info[0])
+                    on_complete = functools.partial(self.installer.reenable_package, info[0])
```

Comment:
Using functools.partial should be commented in some way because right now it is confusing for someone stepping into the codebase. Why is a partial function invocation being created?